src/main/java/btree4j/utils/lang/ArrayUtils.java

```diff
@@ -300,21 +301,29 @@ public static int lastIndex(final Object array) {
      * @throws IndexOutOfBoundsException if the index is out of range (index < 0 || index >=
      *         array.length), or if the array is <code>null</code>.
      * @since 2.1
+     * @implNote Its dependance on removeFrom may slow down the function.
      */
     @SuppressWarnings("unchecked")
     public static <T> T[] remove(final T[] array, final int index) {
+        int length = getLength(array);
+        Object result = Arrays.copyOf(removeFrom(array, index), length - 1);
+        if (index < length - 1) {
+            System.arraycopy(array, index + 1, result, index, length - index - 1);
+        }
+        return (T[]) result;
+    }
+
+    public static <T> T[] removeFrom(final T[] array, final int index) {
```

Comment:
Why not simply use JDK's [copyOf](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#copyOf-T:A-int-)?

Reply:
`removeFrom` sounds that elements are removed from original array. `copyFrom` would be more better naming just using Arrays.copyOfRange as follows:

```
    public static <T> T[] copyFrom(final T[] original, final int from) {
        return Arrays.copyOfRange(original, from, original.length);
    }
```

Reply:
That's right. I didn't know of copyOf. It is doing the function I was looking for.